Frontend/student-hub/src/components/Login.js

```diff
@@ -25,7 +27,23 @@ function Login() {
             // if correct navigate to the dashboard
             if (response.ok) {
                 const data = await response.json();
+                // check for userId in the data and not undefined
+                if (data.userId !== undefined){
+                  //store token and userId in sessionstorgae
                 sessionStorage.setItem('token', data.token);
+
+                sessionStorage.setItem('userId', data.userId)
+
+                // Check if the userId exist in the database
+                const isValidUser = await checkUserId(data.userId);
+                if (isValidUser) {
+                  navigate('/dashboard');
+                } else {
+                  setError('Invalid user credentials.');
+                }
+              } else {
+                console.error('userId is undefined in data:', data);
```

Comment:
let's remove this console.error and instead use the new error toast! or the existing setError